### ISSUE TITLE
Add meditation timer

### ISSUE THREAD

**Issue by liza-veis** [BinaryStudioAcademy/bsa-2023-calmpal#161]
- [ ] add modal where user can select meditation duration
- [ ] add only predefined values for now - 5, 10, 20
- [ ] after user selects an item from meditations list #71 , modal should appear
- [ ] after user clicks start session in the modal, they should be redirected to the meditation player screen #78  and the sound should loop for that duration

Note: we don't have design for now, so just try to stick to the application style. Also we need to add modal in #160 too, so you can cooperate with @Dmytro-Kucherenko 

![Image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/64521658/8343b328-0fd1-4381-8744-4af0ae258029)

**Comment by vladyslavdobychin** [comment]
High-level plan for implementing the modal: 

- Flow should be `Meditation list item > Modal with a timer > Session starts > Meditation player`
- The modal also needed in #160 with a following flow: `Add meditation > Modal`
- At this point, modal should be universal, potentially it should stretch to intake different amount of contact, elements, inputs etc. 
- The completion of issue depends on issues #71 and #78  since the modal is triggered from within functionality of issue #71 and the modal's 'Start' button should direct the user to #78. Both of them are not merged yet, so It would be wise to start implementing the modal as an independent, universal component. 
- Will need to contact @Dmytro-Kucherenko to see how we can test the modal with the data from his issue, so I could make sure the modal is truly universal and can intake all possible sorts of data. 

- [x] Think through the logic of how it should intake different data and still look consistent in all instances.
- [x] Outline basics of the modal: minimum size, shape, basic styling, fonts, colors.
- [ ] Create a timer as a different component, that could be passed to the modal.
- [ ] The timer should include the button, locate an already implemented logic of a button component to be used.
- [ ] Test modal component with Dmytro's data.

**Comment by vladyslavdobychin** [comment]
Currently working on just the modal window without a timer, want to get it done to pass it to other teammates who need it in their issues so we could start testing. 
As of now, locally created a draft for the modal and a testing parent component. I am a bit held back by my primary issues #61 but trying to get back to the modal as often as I can.

**Comment by vladyslavdobychin** [comment]
<img width="464" alt="Screenshot 2023-09-11 at 21 03 08" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/d3e3d8bb-3e61-4424-ad4b-7eef48403b0a">

**Comment by vladyslavdobychin** [comment]
Good evening, @Liza-Veis 
Wanted to ask a quick question: 
So, according to the task definition, when the user selects meditation item, a timer should pop-up. 
What I noticed, is that each meditation item has its own fixed time duration, when the timer appears, should it automatically highlight the default duration as currently selected? 
Example below: 

<img width="500" alt="Screenshot 2023-09-11 at 21 09 26" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/7489b518-a64b-4d08-9c7f-6a0a8d84996f">

<img width="403" alt="Screenshot 2023-09-11 at 21 09 40" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/02ba4ea1-4d55-48ad-8897-3d42b0809a5a">

**Comment by vladyslavdobychin** [comment]
https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/a1c0e00d-31d3-41a2-a7c7-ec067b72ead2

**Comment by KrychkovskaVira** [comment]
@vladyslavdobychin 
Hello! I have several comments regarding the task:
1. The pop-up window is not displaying correctly at resolutions of 360x800 and 390x844


![Image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/88768615/ec540aa7-169b-4f2d-9c97-7e96c335529b)


2. The time is displayed in the format 00:00:00 on the pop-up window;  can it be displayed in minutes only?

**Comment by vladyslavdobychin** [comment]
Hi, @KrychkovskaVira 
1. Will take care of that.
2. Well, it definitely shouldn't display in a format `00:00:00` but something more like this: 
I'm also not sure if it should be just minutes, the implementation from the screenshot was approved by Lisa and Volodymyr, I also think it should be displayed in at least `02:51` format to display the actual length of the audio file. 
But we can check with @Liza-Veis about that.

<img width="488" alt="Screenshot 2023-09-25 at 12 50 44" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/b9179898-6865-48de-969d-94cae5c2ae53">

**Comment by vladyslavdobychin** [comment]
@KrychkovskaVira could also, please, remind me that: 
You will need to create a separate ticket for this issue and only when I assign this ticket to myself, I can create a fix branch?

**Comment by KrychkovskaVira** [comment]
If there are multiple comments, we can work within the scope of this ticket

**Comment by vladyslavdobychin** [comment]
In this case I'll get down to work
